src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,76 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("d.MM.yyyy");
+    private static final int INDEX_DATE = 0;
+    private static final int INDEX_NAME = 1;
+    private static final int INDEX_HOUR = 2;
+    private static final int INDEX_SALARY = 3;
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        Employee[] employees = new Employee[names.length];
+        for (int i = 0; i < names.length; i++) {
+            Employee employee = new Employee(names[i]);
+            for (int j = 0; j < data.length; j++) {
+                if (names[i].contains(extractName(data[j]))) {
+                    if (containDate(extractDate(data[j]), dateFrom, dateTo)) {
+                        employee.addSalary(extractSalary(data[j]), extractHour(data[j]));
+                    }
+                }
+            }
+            employees[i] = employee;
+        }
+        return printData(employees, dateFrom, dateTo);
+    }
+
+    private boolean containDate(String date, String dateFrom, String dateTo) {
+        LocalDate dateFrom1 = convertStringToDate(dateFrom);
+        LocalDate dateTo1 = convertStringToDate(dateTo);
+        LocalDate date1 = convertStringToDate(date);
+        if (!date1.isBefore(dateFrom1) && !date1.isAfter(dateTo1)) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    private LocalDate convertStringToDate(String date) {
+        return LocalDate.parse(date, FORMATTER);
+    }
+
+    private String extractDate(String data) {
+        String[] strings = data.split(" ");
```

Comment:
You're splitting data 4 times in order to extract the date, name, salary, and hours